src/common/utils.hpp

```diff
@@ -36,13 +36,25 @@
 
 #include <string>
 
+#include <assert.h>
+
 #include <commissioner/defines.hpp>
 #include <commissioner/error.hpp>
 
-#define SuccessOrDie(aError)                   \
-    do                                         \
-    {                                          \
-        VerifyOrDie((aError) == Error::kNone); \
+#define ASSERT(aCondition)           \
```

Comment:
I wonder if it follows best practices to allow side-effects in `ASSERT`. It goes against conventions.

Reply:
I think we are not supposed to allowed side-effect in `ASSERT` unless we explicitly document it.
I did remove the comment about allowing side-effect os it is not supporting side-effect and if there are code includes side-effect in `ASSERT`, it should be treated as a bug.

Reply:
aren’t this code allowing side effects?